cognite/client/_api/workflows.py

```diff
@@ -293,6 +293,37 @@ def cancel(self, executions: Sequence[CancelExecution]) -> Sequence[WorkflowExec
 
         return [WorkflowExecution._load(execution) for execution in response.json()["items"]]
 
+    def retry(self, id: str, client_credentials: ClientCredentials | None = None) -> WorkflowExecution:
+        """`Retry a workflow execution. <https://api-docs.cognite.com/20230101-beta/tag/Workflow-executions/operation/WorkflowExecutionRetryn>`_
+
+        Args:
+            id (str): The server-generated id of the workflow execution.
+            client_credentials (ClientCredentials | None): Specific credentials that should be used to retry the workflow execution. When passed will take precedence over the current credentials.
+
+        Returns:
+            WorkflowExecution: The retried workflow execution.
+
+        Examples:
+            Retry a workflow execution that has been cancelled or failed:
+
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import CancelExecution
+                >>> client = CogniteClient()
+                >>> res = client.workflows.executions.trigger("foo", "1")
+                >>> client.workflows.executions.cancel([CancelExecution(res.id, "test cancellation")])
+                >>> client.workflows.executions.retry(res.id)
+        """
+        self._warning.warn()
+        nonce = create_session_and_return_nonce(
+            self._cognite_client, api_name="Workflow API", client_credentials=client_credentials
+        )
+        body = {"authentication": {"nonce": nonce}}
+        response = self._post(
+            url_path=f"{self._RESOURCE_PATH}/{id}/retry",
+            json=body,
```

Comment:
[just a random 🏌️ suggestion]
Since you use multiline for `_post` you might as well move the body:
```suggestion
        response = self._post(
            url_path=f"{self._RESOURCE_PATH}/{id}/retry",
            json={"authentication": {"nonce": nonce}},
```